### ISSUE TITLE
How is XLM different from Multilingual BERT?

### ISSUE THREAD

**Issue by fallingstar621** [facebookresearch/XLM#36]
Great work and article! Just curious how is XLM different from multilingual BERT (https://github.com/google-research/bert/blob/master/multilingual.md) regarding pre-training objectives and methods?

**Comment by aconneau** [comment]
Hi @fallingstar621, as mentioned in the paper there are several differences between our implementation and the one of BERT like the fact that we subsample the outputs, we use streams of text, we use BPE instead of sentencepiece etc. For the multilingual part, the multilingual BERT is MLM-only. Our XLM-MLM is thus very similar to the multilingual BERT. For XNLI, we use TLM in addition to MLM which further boosts the accuracy. Hope this helps! Thanks

**Comment by yilinyang7** [comment]
I believe the main difference is that XLM is additionally trained on the parallel corpus.

**Comment by glample** [comment]
@yilinyang7 yes, only when the TLM objective is involved. Also, we never use the "next sentence prediction" objective.

**Comment by fallingstar621** [comment]
Thank you guys for sharing more insights, it is very helpful!
